apis/config/v1beta1/configuration_types.go

```diff
@@ -61,6 +63,101 @@ type Configuration struct {
 	Integrations *Integrations `json:"integrations,omitempty"`
 }
 
+type ControllerManager struct {
```

Comment:
Sorry for my unclear comment.
I meant adding a link like this:

```suggestion
https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b831fffe7ccc3c8145306c3e1a3b1b14ffa/pkg/config/v1alpha1/types.go#L30
type ControllerManager struct {
```

Also, we need to copy all APIs from the controller-runtime repo to this repo. If we drop some fields, this causes breaking API changes.

Reply:
I don't think I follow, all the deprecated structs are now internal and only used by us.

Reply:
I guess he mean
```golang
// Copied from <url>
```

But also I don't think it's that important, since it's being removed anyways.

Reply:
I meant that we shouldn't remove  `SyncPeriod` and `LeaderElection`, and so on.
We should copy all fields from `ControllerManagerConfigurationSpec` to the kueue.

https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b831fffe7ccc3c8145306c3e1a3b1b14ffa/pkg/config/v1alpha1/types.go#L30-L76

Reply:
Ah, there is `Leaderelection` in L74.

Reply:
But I couldn't yet see `SyncPeriod` and `CacheNamespace` and `GracefulShutdown` and so on in this file.

Reply:
Or, Is it no longer possible to set these fields in ctrl.Manager?

Reply:
`GracefulShutdown` - was never actually used.

`SyncPeriod` and `CacheNamespace` had nor reference in kueue, If we have a concrete use-case in which we need the we can add them , otherwise they are just dead code.

Reply:
@trasc Right. We don't use those parameters. But currently, those don't affect kueue, but users can set those parameters.
If we drop those parameters, don't the APIs broken?
Therefore, can you verify the following steps?

1. Deploy kueue with v0.3.2
2. Update only the image version from v0.3.2 to this commit version.
3. Verify that kueue can load the old scheme config (v0.3.2).

if the above verification works well, I'm ok with dropping unused parameters.

My concern is only that kueue doesn't violate kubernetes API policy: https://kubernetes.io/docs/reference/using-api/deprecation-policy/

cc: @alculquicondor

Reply:
There are no changes in `/config` and the e2e are running, this should cover the scenario described above.

Also, in config_test, with the exception of `fullControllerConfig` which is new, all the other test config  files have the sane content as before.

Reply:
If users set `SyncPeriod`, what happens? If kueue-controller with this commit isn't crashed, I'm ok.

Reply:
Currently , just like in the deprecated implementation any unknown field is ignored. The controller will not crash but  it will have no effect.

Reply:
It makes sense. I'm ok with dropping unused fields. Thanks for the clarifications.